src/main/scala/org/broadinstitute/dsde/workbench/sam/openam/PostgresAccessPolicyDAO.scala

```diff
@@ -237,20 +242,76 @@ class PostgresAccessPolicyDAO(protected val dbRef: DbReference,
     }
   }
 
-  def listResourcesConstrainedByGroup(groupId: WorkbenchGroupIdentity): IO[Set[Resource]] = ???
-  def createPolicy(policy: AccessPolicy): IO[AccessPolicy] = ???
-  def deletePolicy(policy: FullyQualifiedPolicyId): IO[Unit] = ???
-  def loadPolicy(resourceAndPolicyName: FullyQualifiedPolicyId): IO[Option[AccessPolicy]] = ???
-  def overwritePolicyMembers(id: FullyQualifiedPolicyId, memberList: Set[WorkbenchSubject]): IO[Unit] = ???
-  def overwritePolicy(newPolicy: AccessPolicy): IO[AccessPolicy] = ???
-  def listPublicAccessPolicies(resourceTypeName: ResourceTypeName): IO[Stream[ResourceIdAndPolicyName]] = ???
-  def listPublicAccessPolicies(resource: FullyQualifiedResourceId): IO[Stream[AccessPolicy]] = ???
-  def listResourcesWithAuthdomains(resourceTypeName: ResourceTypeName, resourceId: Set[ResourceId]): IO[Set[Resource]] = ???
-  def listResourceWithAuthdomains(resourceId: FullyQualifiedResourceId): IO[Option[Resource]] = ???
-  def listAccessPolicies(resourceTypeName: ResourceTypeName, userId: WorkbenchUserId): IO[Set[ResourceIdAndPolicyName]] = ???
-  def listAccessPolicies(resource: FullyQualifiedResourceId): IO[Stream[AccessPolicy]] = ???
-  def listAccessPoliciesForUser(resource: FullyQualifiedResourceId, user: WorkbenchUserId): IO[Set[AccessPolicy]] = ???
-  def listFlattenedPolicyMembers(policyId: FullyQualifiedPolicyId): IO[Set[WorkbenchUser]] = ???
-  def setPolicyIsPublic(policyId: FullyQualifiedPolicyId, isPublic: Boolean): IO[Unit] = ???
-  def evictIsMemberOfCache(subject: WorkbenchSubject): IO[Unit] = ???
+  override def listResourcesConstrainedByGroup(groupId: WorkbenchGroupIdentity): IO[Set[Resource]] = {
+    import SamTypeBinders._
+
+    runInTransaction { implicit session =>
+      val r = ResourceTable.syntax("r")
+      val ad = AuthDomainTable.syntax("ad")
+      val g = GroupTable.syntax("g")
+      val rt = ResourceTypeTable.syntax("rt")
+
+      val constrainedResourcesPKs = samsqls"""select ${ad.result.resourceId}
+              from ${AuthDomainTable as ad}
+              where ${ad.groupId} = (${workbenchGroupIdentityToGroupPK(groupId)})"""
+
+      val results = samsql"""select ${rt.result.name}, ${r.result.name}, ${g.result.name}
+                      from ${ResourceTable as r}
+                      join ${ResourceTypeTable as rt} on ${r.resourceTypeId} = ${rt.id}
+                      left join ${AuthDomainTable as ad} on ${r.id} = ${ad.resourceId}
+                      left join ${GroupTable as g} on ${ad.groupId} = ${g.id}
+                      where ${r.id} in (${constrainedResourcesPKs})"""
```

Comment:
I don't love that this has a sub-subquery, but I'm not sure how else to do this all in one query

Reply:
i feel like there's a way to do this with more joins and filters. i'm gonna think about this some more...

Reply:
I flattened it to a single subquery, leaving it one level deep